src/components/Test.vue

```diff
@@ -105,11 +114,21 @@ export default {
       document.addEventListener('keydown', this._keyListener.bind(this));
     },
 
+    async updatePreference (params) {
```

Comment:
I suggest to rename "params" to "preference" or "updatedTopic" to know what it is.